### PR TITLE
Add base for AlgorithmPage testing and modify login page according to…

### PR DESCRIPTION
**Pull request by Ana-cpu**
Add base for AlgorithmPage testing and modify login page according to last application version

### CODE REVIEW

File: endavaTraining/src/main/java/com/endavatraining/util/Utils.java [hunk 1]
```diff
@@ -93,6 +95,38 @@ public static LoginPage setUpWebBrowser(String browser) {
...
+			algorithmsPage = new AlgorithmsPage(new OperaDriver());
+		}else if (browser.equalsIgnoreCase("ie")){
+			WebDriverManager.iedriver().setup();
+			algorithmsPage = new AlgorithmsPage(new InternetExplorerDriver());
+		}
+
+		else {
+			throw new RuntimeException();
+		}
+		return algorithmsPage;
```
Comment (jana9281) [inline endavaTraining/src/main/java/com/endavatraining/util/Utils.java]: This is not a good approach  - you should be instantiating driver once per test - not per page!

File: endavaTraining/src/test/java/com/endavatraining/TestAlgorithmsPage.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		algorithmsPage.open();
+		new WebDriverWait(algorithmsPage.driver, 5)
+				.until(ExpectedConditions.visibilityOfElementLocated(algorithmsPage.getSubmitButton()));
+	}
+
+    /*
+     * After Test suite message
+     * @author ana.acanski
+     *
+     */
```
Comment (jana9281) [inline endavaTraining/src/test/java/com/endavatraining/TestAlgorithmsPage.java]: Please remove this comment - no need to set an author multiple times

File: endavaTraining/src/test/resources/testng.xml [hunk 3]
```diff
@@ -54,5 +61,5 @@
 		<classes>
 			<class name="com.endavatraining.TestProfilePage" />
 		</classes>
-	</test>
-</suite> <!-- Suite -->
+	</test> -->
+</suite> <!-- Suite -->
```
Comment (jana9281) [inline endavaTraining/src/test/resources/testng.xml]: Please do not comment out tests in your PRs

File: endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+public class AlgorithmsPage extends BasePage {
+	private static final String ENDALGORITHMS_URL = "http://localhost:8080/algorithms";
+	public static Logger log = Logger.getLogger(AlgorithmsPage.class);
+	private By upperRightLogInButton = By.xpath("//a[@href='/login']");
+	private By formControl = By.id("length");
+	private By searchKey = By.id("plainText");
+	private By vgnKey = By.id("vgnKey");
+	private By letter = By.id("letter");
+	private By algorithmsPageWelcomeMessage = By.xpath("//div[@class='panel-title text-center'][contains(.,'Hello, and welcome to our gamers page!')]");
+    private By submitButton = By.xpath("/html/body/div[1]/div/div/div[2]/div[1]/form/div/div[5]/button");
```
Comment (jana9281) [inline endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java]: Indentation

Comment (jana9281) [inline endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java]: Is this the only way you could locate this Submit button?
This locator is not well written.

File: endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+		super(driver);
+	}
+
+	public void open() {
+		log.debug("Open endava algorithms site");
+		driver.get(ENDALGORITHMS_URL);
+	}
+
+	public By getSubmitButton() {
+
```
Comment (jana9281) [inline endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java]: remove empty line

File: endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+		return submitButton;
+
+	}
+
+
+    public void insertValueInFormControlFields(int formControl) {
+        driver.findElement(this.formControl).sendKeys("6");
```
Comment (jana9281) [inline endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java]: This shouldn't be hardcoded to 6, right?

File: endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+		log.debug("Inserted number in form control number field");
+    }
+
+    public void insertValueInSearchKeyFields(String searchKey) {
```
Comment (jana9281) [inline endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java]: same hardcoding here

File: endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+        driver.findElement(this.searchKey).sendKeys("ana");
+		log.debug("Inserted searchKey in search key field");
+    }
+
+    public void insertValueInVigenereFields(String vgnKey) {
```
Comment (jana9281) [inline endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java]: same

File: endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+        driver.findElement(this.vgnKey).sendKeys("FJv6Jb");
+		log.debug("Inserted FJv6Jb in Vigenere key field");
+    }
+
+    public void insertValueInLetterFields(char letter) {
```
Comment (jana9281) [inline endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java]: same

File: endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+        driver.findElement(submitButton).click();
+		log.debug("Clicked on Submit button");
+    }
+    
+    public void submitNewValue(int formControl, String searchKey, String vgnKey, char letter) {
+
+        driver.findElement(this.formControl).sendKeys("6");
+        driver.findElement(this.searchKey).sendKeys("ana");
+        driver.findElement(this.vgnKey).sendKeys("FJv6Jb");
+        driver.findElement(this.letter).sendKeys("e");
```
Comment (jana9281) [inline endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java]: shouldn't you be using some Utils methods here?

File: endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+public class AlgorithmsPage extends BasePage {
+	private static final String ENDALGORITHMS_URL = "http://localhost:8080/algorithms";
+	public static Logger log = Logger.getLogger(AlgorithmsPage.class);
+	private By upperRightLogInButton = By.xpath("//a[@href='/login']");
+	private By formControl = By.id("length");
+	private By searchKey = By.id("plainText");
+	private By vgnKey = By.id("vgnKey");
+	private By letter = By.id("letter");
+	private By algorithmsPageWelcomeMessage = By.xpath("//div[@class='panel-title text-center'][contains(.,'Hello, and welcome to our gamers page!')]");
```
Comment (jana9281) [inline endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java]: Let's see if there is a better locator for this label

File: endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java (hunk 10 above)
Comment (Ana-cpu) [reply endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java]: this method contains clickSubmitButton();
I didn't find appropriate Utils methods, maybe I will update this part later according to Test suite updating.

File: endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java (hunk 11 above)
Comment (Ana-cpu) [reply endavaTraining/src/main/java/com/endavatraining/pages/AlgorithmsPage.java]: I think that we can keep this locator for title label. it is already used in the other test case.
I will think about this suggestion, and if it is possible  i will update this part later according to Test suite updating if you agree.

**Comment by Ana-cpu** [comment]
all suggested comments are updated and according to task T3_ana.acanski (test environment refactoring) I will close this task and remove this branch

**Pull request closed without merging**